File: budget/budget-api-v2/domain/money.go
package domain

import "github.com/shopspring/decimal"

var SCALE_PRECISION = 2
var ZERO, _ = MoneyFor("0.00")
var ONE, _ = MoneyFor("1.00")

type Money struct {
	content decimal.Decimal
}

func MoneyFor(amount string) (*Money, error) {
	return nil, nil
}

func (m *Money) Plus(money *Money) *Money {
	return nil
}

func (m *Money) StringifyAmount() string {
	return ""
}

File: budget/analytic-api/src/app/money/domain/money.py
from dataclasses import dataclass
from decimal import ROUND_HALF_DOWN, Context, Decimal
from typing import Final

SCALE_PRECISION: Final = 2
SCALE_CRITERIA: Final = ROUND_HALF_DOWN

MONEY_ARITHMETIC_CONTEXT: Final = Context(rounding=SCALE_CRITERIA)


@dataclass
class Money:
    amount: Decimal

    @staticmethod
    def money_for(amount: str) -> "Money":
        context = MONEY_ARITHMETIC_CONTEXT
        return Money(context.create_decimal(Decimal(amount)))

    def stringify_amount(self) -> str:
        return str(self.amount)

    def plus(self, addend: "Money") -> Decimal:
        context = MONEY_ARITHMETIC_CONTEXT

        first_amount = self.amount
        second_amount = addend.amount
        result = round(first_amount + second_amount, SCALE_PRECISION)
        return context.create_decimal(result)
